lang/java/reef-tang/tang/src/main/java/org/apache/reef/tang/implementation/java/JavaNodeFactory.java

```diff
@@ -142,25 +142,27 @@
   /**
    * XXX: This method assumes that all generic types have exactly one type parameter.
    */
-  public static <T> NamedParameterNode<T> createNamedParameterNode(Node parent,
-                                                                   Class<? extends Name<T>> clazz, Type argClass)
+  public static <T> NamedParameterNode<T> createNamedParameterNode(final Node parent,
+                                                                   final Class<? extends Name<T>> clazz,
+                                                                   final Type argClass)
       throws ClassHierarchyException {
 
-    Class<?> argRawClass = ReflectionUtilities.getRawClass(argClass);
+    Type argClazz = argClass;
```

Comment:
How about we make this `final` (without assigning it), then assign it within the `if (isSet...)` block and an additional `else` block. A more descriptive name would be better, but what that is escapes me.